README.md

```diff
@@ -17,6 +17,122 @@ Comming soon ;)
 
 ## Configuration
 
+### Basic Configuration
+The basic configuration includes all the minimal configuration to deploy the app successfully. Create a file named `boss.yml` in your project's root directory. The file should include the following:
+
+```yml
+#boss.yml
+project_name: PROJECT_NAME
+project_description: PROJECT_DESCRIPTION
+repository_url: https://github.com/<username>/<project-name>
+user: USERNAME
+port: SERVER_PORT
+```
+
+### Stages
+This includes different instances of your application like `dev`, `qa`, `uat`, `staging`, `production`.
+
+```yml
+#boss.yml
+...
+stages:
+    dev:
+        host: dev.your-app.com
+        public_url: http://dev.your-app.com
+        app_dir: /path/to/your/app
+        logging:
+            files:
+                - /path/to/error/log/file
+                - /path/to/access/log/file
+    uat:
+        host: uat.your-app.com
+        public_url: http://uat.your-app.com
+        app_dir: /path/to/your/app
+        logging:
+            files:
+                - /path/to/error/log/file
+                - /path/to/access/log/file
+    production:
+        host: your-app.com
+        port: PRODUCTION_SERVER_SSH_PORT
+        username: PRODUCTION_SERVER_USERNAME
+        public_url: http://your-app.com
+        app_dir: /path/to/your/app
+        logging:
+            files:
+                - /path/to/error/log/file
+                - /path/to/access/log/file
+```
+
+### Notifications
+This includes the settings for notifying on deployment start and completion. Currently, only hipchat and slack are supported. Further integrations are welcome through pull-requests.
```

Comment:
`This includes the ...` has been a pattern for each section. :smile: . It'd be better if we can have something different.